warp/yul/main.py

```diff
@@ -89,6 +91,49 @@ def transpile_from_yul(yul_ast: ast.Node) -> str:
     )
 
 
+def parse_to_normalized_ast(codes) -> ast.Node:
+    deployment_code = parse_node(codes["deploymentCode"])
+    runtime_code = parse_node(codes["runtimeCode"])
+    assert isinstance(deployment_code, ast.Block)
+    assert isinstance(runtime_code, ast.Block)
+
+    ctor_block, deployment_functions = extract_top_level_code(deployment_code)
+    main_block, runtime_functions = extract_top_level_code(runtime_code)
+    ctor = ast.FunctionDefinition(
+        name="__constructor_meat", parameters=[], return_variables=[], body=ctor_block
+    )
+    main_ = ast.FunctionDefinition(
+        name="__main_meat",
+        parameters=[],
+        return_variables=[],
+        body=main_block,
+    )
+
+    deployment_names = {
+        x.name for x in deployment_functions if isinstance(x, ast.FunctionDefinition)
+    }
+    runtime_names = {
+        x.name for x in runtime_functions if isinstance(x, ast.FunctionDefinition)
+    }
+    duplicated = deployment_names & runtime_names
+    renamer = Renamer(lambda x: x + "_deployment" if x in duplicated else x)
```

Comment:
Could you write a test for this case?

Reply:
Hmm, I believe it is tested by some of the current tests, because before I wrote that "rewrite" piece they didn't pass because of function duplication. I can write a specific test-case (but it would require a little changing)

Reply:
If it fixes the other I think it aught to be fine

Reply:
This change is not actually related to this branch, it's just that I haven't rebased it yet. I've written a test in #142